pkg/controller/apiextensions/composite/composed/api.go

```diff
@@ -25,16 +25,24 @@ import (
 	"k8s.io/apimachinery/pkg/util/json"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
 	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
 	"github.com/crossplane/crossplane-runtime/pkg/resource"
 
 	"github.com/crossplane/crossplane/apis/apiextensions/v1alpha1"
 )
 
 const (
-	errUnmarshal = "cannot unmarshal base template"
-	errFmtPatch  = "cannot apply the patch at index %d"
-	errGetSecret = "cannot get connection secret of composed resource"
+	errUnmarshal  = "cannot unmarshal base template"
+	errFmtPatch   = "cannot apply the patch at index %d"
+	errGetSecret  = "cannot get connection secret of composed resource"
+	errNamePrefix = "name prefix is not found in labels"
+)
+
+const (
+	// LabelKeyNamePrefixForComposed is the prefix that will be used by composed
+	// resources.
+	LabelKeyNamePrefixForComposed = "crossplane.io/root-composite"
```

Comment:
I'm on the fence about the value of this. `composite` didn't make much sense to me as the leafs are actually member of multiple composites but what we'd like to say it's the one that started it all. Made it `name-prefix` at one point and that's what it is technically but didn't seem like a very good name from UX perspective.

Reply:
I lean toward keeping this as just `crossplane.io/composite` and living with the fact that it will always be the outermost composite. I'm not yet sure how frequent nested composites will even be. That said, I don't feel super strongly about the name.

Reply:
> That said, I don't feel super strongly about the name.

Yeah, me neither. I'll just make it `crossplane.io/composite`